src/vs/workbench/contrib/terminal/common/terminalService.ts

```diff
@@ -422,10 +422,10 @@ export abstract class TerminalService implements ITerminalService {
 	}
 
 	protected _validateShellPaths(label: string, potentialPaths: string[]): Promise<[string, string] | null> {
-		if (potentialPaths.length === 0) {
+		let current: string;
+		if (potentialPaths.length === 0 || (current = potentialPaths.shift()!) === '') {
 			return Promise.resolve(null);
 		}
-		const current = potentialPaths.shift();
 		return this._fileService.exists(URI.file(current!)).then(exists => {
```

Comment:
I think `!` of this line needs to be removed.
Because the value to which `!` is applied is already in `current 'by `current = potentialPaths.shift()!`.

Thank you for implementing my proposal!